master/internal/experiment/bulk_action.go

```diff
@@ -44,6 +44,28 @@ type deleteExperimentOKResult struct {
 // ExperimentsAddr is the address to direct experiment actions.
 var ExperimentsAddr = actor.Addr("experiments")
 
+// terminalExperimentStates where an experiment can be deleted, archived, or unarchived.
+var terminalExperimentStates = func() []string {
+	states := make([]string, len(model.TerminalStates))
+	idx := 0
+	for state := range model.TerminalStates {
+		states[idx] = string(state)
+		idx++
+	}
+	return states
+}()
+
+// nonTerminalStates where an experiment can be canceled or killed.
+var nonTerminalStates = func() []experimentv1.State {
+	var states []experimentv1.State
+	for s := range model.ExperimentTransitions {
+		if !model.TerminalStates[s] && !strings.Contains(string(s), "DELET") {
```

Comment:
```suggestion
		if !model.TerminalStates[s] && !strings.Contains(string(s), "DELET") {
```
just add a const for 'deleting states'. also deleting states probably should be considered terminal.. but that may have unintended consequences.

Reply:
also i would put this into model as constants

Reply:
ok, I think latest commit does this. Note that I needed to remove delete states to make the SQL query happy (avoiding state or states not in postgres's experimentstate enum)